src/gui_common/menus/PlanetSettings.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Globalization;
+using Godot;
+
+/// <summary>
+///   Planet settings in the planet customizer.
+/// </summary>
+public partial class PlanetSettings : VBoxContainer
+{
+    [Export]
+    private OptionButton lifeOriginButton = null!;
+
+    [Export]
+    private OptionButton worldSizeButton = null!;
+
+    [Export]
+    private OptionButton worldTemperatureButton = null!;
+
+    [Export]
+    private OptionButton worldSeaLevelButton = null!;
+
+    [Export]
+    private OptionButton worldGeologicalActivityButton = null!;
+
+    [Export]
+    private OptionButton worldClimateInstabilityButton = null!;
+
+    [Export]
+    private CheckButton lawkOnlyButton = null!;
+
+    [Export]
+    private CheckButton dayNightCycleButton = null!;
+
+    [Export]
+    private HSlider dayLength = null!;
+
+    [Export]
+    private LineEdit dayLengthReadout = null!;
+
+    [Export]
+    private VBoxContainer dayLengthContainer = null!;
+
+    [Signal]
+    public delegate void LawkSettingsChangedEventHandler(bool lawkOnly);
+
+    [Signal]
+    public delegate void LifeOriginSettingsChangedEventHandler(int lifeOrigin);
```

Comment:
As this is exposed as a signal from a comment, I think this needs to be refactored to use a proper enum (or string values) so that GUI indexes aren't passed between components. To me that seems very brittle and against the principal of making GUI components' interfaces insulated from their internal implementation details.